controllers/metallb_controller.go

```diff
@@ -244,6 +248,15 @@ func (r *MetalLBReconciler) syncMetalLBResources(config *metallbv1beta1.MetalLB)
 			}
 		}
 
+		if obj.GetKind() == "Secret" && obj.GetName() == MetalLBWebhookSecret {
+			secret := &v1.Secret{}
+			err := r.Client.Get(context.TODO(), types.NamespacedName{Name: MetalLBWebhookSecret, Namespace: obj.GetNamespace()}, secret)
+			if err == nil && obj != nil {
```

Comment:
obj here can't be nil, otherwise we'd throw a nil pointer in the if body.
Also, I'd check the error because we may want to exit if the err is not nil but is not "not found".

Reply:
right, i was planning to check the secret is not nil

Reply:
but secret can't be nil either I guess?

Reply:
you mean if the err is nil it found the requested secret?

Reply:
I mean, nil is a pointer that is initialized to &v1.Secret{}. The function won't change the value of the pointer, so it will continue to point `&v1.Secret{}` regardless of success of failure.

Reply:
right, so i guess checking the error is enough

Reply:
yes i will add the notfound. my comment was regarding the first part